tasks/sync-granule/GranuleFetcher.js

```diff
@@ -324,33 +430,65 @@ class GranuleFetcher {
 
     // if not enabled, make it enabled
     if (versioning.Status !== 'Enabled') {
-      s3().putBucketVersioning({
-        Bucket: bucket,
-        VersioningConfiguration: { Status: 'Enabled' },
-      }).promise();
+      s3()
+        .putBucketVersioning({
+          Bucket: bucket,
+          VersioningConfiguration: { Status: 'Enabled' },
+        })
+        .promise();
     }
   }
 
   /**
    * Ingest individual files
    *
    * @private
-   * @param {Object} file - file to download
-   * @param {string} destinationBucket - bucket to put file in
-   * @param {string} duplicateHandling - how to handle duplicate files
+   * @param {Object} params - parameters
+   * @param {SyncGranuleFileWithChecksums} params.file - file to download
+   * @param {string} params.destinationBucket - bucket to put file in
+   * @param {DuplicateHandling} params.duplicateHandling - how to handle duplicate files
+   * @param {string} params.collectionId - collection ID
    * value can be
    * 'error' to throw an error,
    * 'replace' to replace the duplicate,
    * 'skip' to skip duplicate,
    * 'version' to keep both files if they have different checksums
-   * @returns {Array<Object>} returns the staged file and the renamed existing duplicates if any
-   */
-  async ingestFile(file, destinationBucket, duplicateHandling) {
+   * @param {string} [params.pathId] - ID (nominally a granuleId) to use as a per-granule prefix
+   * differentiation
+  * @returns {Promise<{
+  *   files: Array<{
+  *     bucket: string,
+  *     key: string,
+  *     source: string,
+  *     size?: number,
+  *     fileName: string,
+  *     type?: string,
+  *     checksumType?: string,
+  *     checksum?: string | number
+  *   }>,
+  *   duplicate: {bucket: string, key: string} | undefined
+  * }>}
+  * - Returns the staged file and the renamed existing duplicates if any
+  */
+  async _ingestFile({
+    file,
+    destinationBucket,
+    duplicateHandling,
+    pathId,
```

Comment:
nit: could this be granuleId for easier parsing of wheredidjacomefromwheredidjago?

Reply:
It could.... given we're hashing it in the method, the distinction isn't as valuable as it was in prior implementations.  I'll make that update.

Reply:
137dbda73ca

Reply:
yea, it's a small thing but I find it more understandable what's happening